### ISSUE TITLE
Issue with Query Depth Limitation in Filters with Multiple Groups

### ISSUE THREAD

**Issue by hsaifurrehman** [directus/directus#24190]
### Describe the Bug

When applying filters with 4 or more groups using "AND" or "OR" logical operations in Directus, the API returns an error stating that the query is invalid. However, the root cause lies in the query parsing depth configuration within the Directus API code.

In the api/src/app.ts file, the query parser depth is explicitly set to 10:

`app.set('query parser', (str: string) => qs.parse(str, { depth: 10 }));`

Due to this limitation, any filter structure exceeding a depth of 10 gets improperly parsed, causing the remaining parts of the filter to be merged into a single string. This results in an invalid query and leads to the error.

**Expected Behavior**: The API should handle deeper filter structures seamlessly without merging or truncating parts of the query.

**Proposed Solution**:  Increase the depth parameter in the query parser configuration to accommodate more complex filter structures. Alternatively, provide a configurable option to allow users to adjust the depth based on their requirements.

This adjustment will ensure that complex queries are parsed correctly, improving the reliability and flexibility of the API.

### To Reproduce

1.  Create a filter with more than 4 nested groups using "AND" or "OR" operations.
2. Send the request to the Directus API.
3. Observe the error indicating an invalid query.

### Directus Version

v10.13.1

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

PostgreSQL

**Comment by rijkvanzanten** [comment]
Can you share an example of a filter that gets that deep? The limit of 10 is indeed a bit arbitrary, but it's there to make sure somebody doesn't accidentally blow up their server by performing queries that are impossibly hard to parse.

**Comment by rijkvanzanten** [comment]
I'm also not entirely sure what a good number would be then 🤔 We've been running with 10 so far and this is the first time that's reported as too limiting, so I'm very curious to see what your use case is to figure out what an appropriate depth might look like
